python/katana/_property_graph.pyx

```diff
@@ -336,32 +274,83 @@ cdef class PropertyGraph:
-    return self.topology().num_nodes()
-{% endcall %}
-{% call numba.method_with_body("num_edges", "uint64_t", []) %}
-    return self.topology().num_edges()
-{% endcall %}
-{% call numba.method_with_body("edge_index", "uint64_t", ["uint64_t"]) %}
-    return self.topology().out_indices.get().Value(arg1)
-{% endcall %}
-{% call numba.method_with_body("get_edge_dest", "uint64_t", ["uint64_t"]) %}
-    return self.topology().out_dests.get().Value(arg1)
-{% endcall %}
-{% endcall %}
-
-{{numba.register_all_wrappers()}}
+        return <uint64_t>self.underlying_property_graph()
+
+
+cdef class PropertyGraph(PropertyGraphBase):
+    """
+    A property graph loaded into memory.
+    """
+
+    cdef _PropertyGraph * underlying_property_graph(self) nogil except NULL:
+        return self._underlying_property_graph.get()
+
+    def __init__(self, path, node_properties=None, edge_properties=None, partition_id_to_load=None):
+        """
+        __init__(self, path, node_properties=None, edge_properties=None, partition_id_to_load=None)
+
+        Load a property graph.
+
+        :param path: the path or URL from which to load the graph. This support local paths or s3 URLs.
+        :type path: str
+        :param partition_id_to_load: The partition number in the graph to load. If this is None (default), then the
+            partition corresponding to this host's network id will be loaded.
+        :param node_properties: A list of node property names to load into memory. If this is None (default), then all
+            properties are loaded.
+        :param edge_properties: A list of edge property names to load into memory. If this is None (default), then all
+            properties are loaded.
+        """
+        cdef RDGLoadOptions opts
+        cdef vector[string] node_props
+        cdef vector[string] edge_props
+        # we need to generate a reference to a uint32_t in order to construct an
+        # optional[uint32_t] in opts
+        cdef uint32_t part_to_load
+        if partition_id_to_load is not None:
+            part_to_load = partition_id_to_load
+            opts.partition_id_to_load = part_to_load
+        if node_properties is not None:
+            node_props = _convert_string_list(node_properties)
+            opts.node_properties = &node_props
+        if edge_properties is not None:
+            edge_props = _convert_string_list(edge_properties)
+            opts.edge_properties = &edge_props
+        path_str = <string>bytes(str(path), "utf-8")
+        with nogil:
+            self._underlying_property_graph = handle_result_PropertyGraph(_PropertyGraph.Make(path_str, opts))
+
+    @staticmethod
+    cdef PropertyGraph make(shared_ptr[_PropertyGraph] u):
+        f = <PropertyGraph>PropertyGraph.__new__(PropertyGraph)
+        f._underlying_property_graph = u
+        return f
+
+    @staticmethod
+    def from_graphml(path, uint64_t chunk_size=25000):
+        """
+        Load a GraphML file into katana form
```

Comment:
qq: we still support GraphML?

Reply:
I have no idea. I was asked to add this by @bowu and there was code implementing it in the graph-convert tools.

Reply:
It's an ask from Intuit.